homeassistant/components/image_processing/clarifai_general.py

```diff
@@ -0,0 +1,141 @@
+"""
+import base64
+import json
+import logging
+
+import voluptuous as vol
+
+from homeassistant.core import split_entity_id
+import homeassistant.helpers.config_validation as cv
+from homeassistant.components.image_processing import (
+    PLATFORM_SCHEMA, ImageProcessingEntity, CONF_SOURCE, CONF_ENTITY_ID,
+    CONF_NAME)
+from homeassistant.const import STATE_UNKNOWN
+
+
+_LOGGER = logging.getLogger(__name__)
+
+CLASSIFIER = 'Clarifai'
+CONF_API_KEY = 'api_key'
+
+REQUIREMENTS = ['clarifai==2.4.1']
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_API_KEY): cv.string,
+})
+
+
+def encode_image(image):
+    """base64 encode an image stream."""
+    base64_img = base64.b64encode(image)
+    return base64_img
+
+
+def parse_concepts(api_concepts):
+    """Parse the API concepts data."""
+    return {concept['name']: round(100.0*concept['value'], 2)
+            for concept in api_concepts}
+
+
+def validate_api_key(api_key):
+    """Check that an API key is valid, if yes return the app."""
+    try:
+        from clarifai.rest import ClarifaiApp, ApiError
+        app = ClarifaiApp(api_key=api_key)
+        return app
+    except ApiError as exc:
+        error = json.loads(exc.response.content)
+        _LOGGER.error(
+            "%s error: %s", CLASSIFIER, error['status']['description'])
+        return None
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the Clarifai classifier."""
+    api_key = config[CONF_API_KEY]
+    app = validate_api_key(api_key)
+    if app is None:
+        return
+
+    entities = []
+    for camera in config[CONF_SOURCE]:
+        entities.append(ClarifaiClassifier(
+            app,
+            camera[CONF_ENTITY_ID],
+            camera.get(CONF_NAME),
+        ))
+    add_devices(entities)
+
+
+class ClarifaiClassifier(ImageProcessingEntity):
+    """Perform a classification via Clarifai."""
+
+    def __init__(self, app, camera_entity, name=None):
+        """Init with the API key."""
+        model = 'general-v1.3'
+        self.model = app.models.get(model)
+        if name:  # Since name is optional.
+            self._name = name
+        else:
+            entity_name = split_entity_id(camera_entity)[1]
+            self._name = "{} {}".format(CLASSIFIER, entity_name)
+        self._camera_entity = camera_entity
+        self._classifications = {}  # The dict of classifications.
+        self._state = STATE_UNKNOWN  # The most likely classification.
+
+    def model_prediction(self, image):
+        """Make a prediction based on an image."""
+        try:
+            from clarifai.rest import ApiError
+            response = self.model.predict_by_base64(encode_image(image))
+            if response['status']['description'] == 'Ok':
+                return response
+        except ApiError:
+            _LOGGER.error("%s error: check your internet connection",
+                          CLASSIFIER)
```

Comment:
Here I would log `str(exc)` as well (same as when constructing `ClarifaiApp`). There are many reasons why this could fail - e.g. maybe the image cannot be decoded on backend, you're out of free requests for the month, etc.

Reply:
updated

Reply:
@robmarkcole Yes, because in that case there's no response received.